### ISSUE TITLE
TensorRT & YOLO 11

### ISSUE THREAD

**Issue by mldl4** [ultralytics/ultralytics#18161]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi! I'm back with problems.

I'm trying to inference a pretrained YOLO11n into a jetson nano. My process follows next steps:
1. I convert the model from pt to onnx using ultralytics
```
    model.export(format=export_to, device=0, half=True, data="visdrone.yaml", task='detect')  # creates 'yolo.engine'
```
2. I transfer to jetson and I convert to engine using tensorrt
```
MODELS_PATH="/home/sts/Documents/YOLO"
MODEL_PATH="${MODELS_PATH}/yolov8.onnx"
SAVE_PATH="${MODELS_PATH}/v8_engine_fp16_int32.trt"
INPUT_NAME="images"
INPUT_SHAPE="1x3x640x640"
PRECISION="fp16"

cd /usr/src/tensorrt/bin
./trtexec --onnx=$MODEL_PATH --saveEngine=$SAVE_PATH --$PRECISION \
 --useDLACore=0 --allowGPUFallback --useSpinWait --buildOnly # --shapes="${INPUT_NAME}:${INPUT_SHAPE}"
cd /home/sts/Documents/YOLO

exit 0
```

3. Then I infer with generated model but predictions (labels and bbox) are not correct.

I don't know what is my error. I have installed different versions of cuda, python and SO.

thanks in advance :)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mldl4, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for guidance, where you can find various [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples to help you get started. Many common questions are addressed there as well!

If this is a 🐛 Bug Report, please ensure to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and troubleshoot the issue. Specifically, include details about your environment (OS, CUDA, TensorRT versions, etc.) and the exact commands or code you are running. Logs or error outputs are also super helpful!

If this is related to a custom ❓ Question, please share more details about your setup, including your dataset, model configuration, and inference results. We also recommend reviewing our [Export Guide](https://docs.ultralytics.com/modes/export/) and [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure a smooth workflow.

## Upgrade

As part of debugging, verify you're using the latest `ultralytics` package along with all its [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) as follows:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run effectively in these verified environments, which come pre-installed with [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Jetson Nano setup**: Ensure you have installed compatible versions of CUDA, TensorRT, and PyTorch for Jetson on your device (this is a common source of compatibility issues).
- **Google Cloud** Deep Learning VM. See the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **Amazon** Deep Learning AMI. See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker Image**. See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/). <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Additional Suggestions

- Verify your input shapes match your model's `--shapes` requirement during TensorRT conversion.
- If predictions are inaccurate, debug using the ONNX model to verify correctness before converting to TensorRT.
- Use precision options (`fp16`, `int8`) cautiously based on your hardware capabilities. Testing with full (`fp32`) precision may help isolate conversion issues.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response 🤖 to help get you started. An Ultralytics engineer will assist you further as soon as possible. Stay tuned! 😊

**Comment by Y-T-G** [comment]
You shouldn't use `half=True` if you want to convert the ONNX to TensorRT.

**Comment by mldl4** [comment]
That's correct, It worked! Thank you very much for you response!!!

**Comment by glenn-jocher** [comment]
@mldl4 you're welcome! Glad to hear it worked. Credit goes to the YOLO community and the Ultralytics team for the tools and resources. Happy coding!
